### DISCUSSION TITLE
vcpkg bootstrap logs to stderr if it downloads cmake or ninja

### DISCUSSION THREAD

**Discussion by DSMitten** [microsoft/vcpkg#19100] [Ideas]
**Describe the bug**
I'm running vcpkg in Azure DevOps with a Microsoft-Hosted agent. vcpkg is a submodule of my team's repo, so we don't use the preinstalled instance on the agent machine - we use the submodule.

When building on the Windows agent, bootstrap succeeds with no errors.
When building on the Mac & Linux agent, bootstrap succeeds but first logs the following:
```
Downloading cmake...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
100   635  100   635    0     0    282      0  0:00:02  0:00:02 --:--:--   282

 43 36.1M   43 15.7M    0     0  5495k      0  0:00:06  0:00:02  0:00:04 5495k
 72 36.1M   72 26.2M    0     0  6767k      0  0:00:05  0:00:03  0:00:02 10.1M
100 36.1M  100 36.1M    0     0  7556k      0  0:00:04  0:00:04 --:--:-- 10.3M
Downloading cmake... done.
Extracting cmake...
Extracting cmake... done.
Downloading ninja...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   617  100   617    0     0   1695      0 --:--:-- --:--:-- --:--:--  1695

100  114k  100  114k    0     0   206k      0 --:--:-- --:--:-- --:--:--  206k
Downloading ninja... done.
Extracting ninja...
Extracting ninja... done.

```

The problem is that the progress messages are logged to stderr. I'm calling vcpkg from a pwsh task, and have configured the task to fail if any errors are written, so writing to the error stream causes the task to fail.

It would be nice if vcpkg could only write to stdout unless there is a problem.

**Comment by DSMitten** [comment]
[pipeline-log.txt](https://github.com/microsoft/vcpkg/files/6094030/pipeline-log.txt)

Attaching log file from my team's ADO pipeline.

**Comment by JackBoosY** [comment]
cc @ras0219 @ras0219-msft

**Comment by BillyONeal** [comment]
curl, which we use here, prints this output to stderr because it wants stdout to be usable as the actual output by various tools; stdout is usually inappropriate for progress notifications.

As a workaround for now you can run bootstrap in a separate task with fail on stderr set to false, or you can redirect stderr in the invoking powershell script.

(Note that boostrap doesn't generally report errors to stderr so looking for it is not very helpful here)

Doing more binary releases of the vcpkg binary itself if/when we figure out the right way to do that for *nix and macOS will help
